container/src/constants/communication.ts

```diff
@@ -1,176 +1,413 @@
+import type {
+  AlertRequestPayload,
+  ConfirmationModalRequestPayload,
+  CurrentRouteRequestPayload,
+  ModalPathDataRequestPayload,
+  ModalSettingsRequestPayload,
+  NavigationRequestPayload,
+  ParamsRequestPayload
+} from 'EventPayloads';
+
 /* eslint-disable @typescript-eslint/no-unsafe-function-type */
 export namespace Events {
   /**
    * Event fired when the micro frontend sends a custom message.
+   * It's dispatched with custom data.
+   * @type {Object}
+   * @property {string} id - event ID
+   * @property {object} data - event data
+   * @returns {void}
+   *
+   * <br><br>
    */
   export const CUSTOM_MESSAGE = 'custom-message';
 
   /**
-   * Event fired when the micro frontend requests the context data.
+   * Event left due to historical reasons - do not use.
+   * @deprecated
+   * @ignore
+   *
+   * <br><br>
    */
   export const GET_CONTEXT_REQUEST = 'get-context-request';
 
   /**
    * Event fired when a navigation has been requested by the micro frontend.
+   * It's dispatched with custom data shown below (check {@link https://github.com/SAP/luigi/blob/main/container/typings/constants/event-payloads.ts|NavigationRequestPayload} details) - some properties are optional.
+   * @type {NavigationRequestPayload}
+   * @example
+   * {
+   *  fromClosestContext: false,
+   *  fromContext: null,
+   *  fromParent: true,
+   *  fromVirtualTreeRoot: false,
+   *  link: '/test/route',
+   *  nodeParams: {}
+   * }
+   * @returns {void}
+   *
+   * <br><br>
    */
   export const NAVIGATION_REQUEST = 'navigation-request';
 
   /**
    * Event fired when the micro frontend requests to show an alert.
+   * It's dispatched with custom data and the payload shown below - some properties are optional. Read more about `showAlert` params [here](https://docs.luigi-project.io/docs/luigi-core-api?section=showalert).
+   * @type {Object.<string, unknown | AlertRequestPayload>}
+   * @property {AlertRequestPayload} payload - event payload (check {@link https://github.com/SAP/luigi/blob/main/container/typings/constants/event-payloads.ts|AlertRequestPayload} details)
+   * @example
+   * payload: {
+   *  text: 'Custom alert message',
+   *  type: 'info',
+   *  links: {
+   *   goToHome: { text: 'Homepage', url: '/overview' },
+   *   goToOtherProject: { text: 'Other project', url: '/projects/pr2' },
+   *   relativePath: { text: 'Hide side nav', url: 'hideSideNav' },
+   *   neverShowItAgain: { text: 'Never show it again', dismissKey: 'neverShowItAgain' }
+   *  },
+   *  closeAfter: 3000
+   * }
+   * @returns {void}
+   *
+   * <br><br>
    */
   export const ALERT_REQUEST = 'show-alert-request';
 
   /**
    * Event left due to historical reasons - do not use.
    * @deprecated
+   * @ignore
+   *
+   * <br><br>
    */
   export const ALERT_CLOSED = 'close-alert-request';
 
   /**
    * Event fired when the micro frontend has been initialized.
+   * It's dispatched with custom data.
+   * @type {Object}
```

Comment:
Since we don't offer any payload which can be used we should not mention it in our docu.